Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -2005,4 +2005,22 @@ class HTTPClientTests: XCTestCase {
 
         self.defaultClient = nil // so it doesn't get shut down again.
     }
+
+    func testTaskPromiseBoundToEL() throws {
+        let elg = getDefaultEventLoopGroup(numberOfThreads: 2)
+        let el = elg.next()
+
+        let client = HTTPClient(eventLoopGroupProvider: .shared(elg))
+
+        defer {
+            XCTAssertNoThrow(try client.syncShutdown())
+            XCTAssertNoThrow(try elg.syncShutdownGracefully())
+        }
+
+        let request = try HTTPClient.Request(url: self.defaultHTTPBinURLPrefix + "/get")
+        let delegate = ResponseAccumulator(request: request)
+        let task = client.execute(request: request, delegate: delegate, eventLoop: .delegate(on: el))
```

Comment:
shouldn't we use `.testOnly(delegateOn: el, channelOn: otherEL)` here to _force_ a foreign channel?

Reply:
ah, you are right, done, thanks!